@W-18905383@ Decouple lint and bundle size GA workflows

adamraya opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you're addressing, a list of required dependencies (if applicable), and any other relevant context. -->

This PR decouples linting and bundle size checks from the main test workflow. Previously, a single workflow handled all checks, which meant linting failures were blocking the entire test run.

Now we have two new independent workflows:
- Lint workflow: Handles linting code for the quality checks
- Bundle size workflow: Validates that bundle sizes is within the defined limits

This separation allows tests to provide feedback even when linting or bundle size checks fail, improving the developer experience and CI efficiency.

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [x] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Created separate `.github/workflows/lint.yml` workflow for linting checks
- Created separate `.github/workflows/bundle-size.yml` workflow for bundle size validation
- Removed linting and bundle size steps from main test workflow


# How to Test-Drive This PR

- Create a PR with linting errors and verify that tests still run
- Create a PR that exceeds the bundle size limits and verify that tests continue
- Check that all three workflows run independently on PR creation
- Verify that all workflows are properly configured as required status checks

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

## General

- [x] Changes are covered by test cases
- [x] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [x] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

adamraya committed:
Extract lint and bundle size workflows

cc-prodsec commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/50848b17-5d32-42ad-9219-cce4e7fc31e7?refs=542cfff737bcd79bb89ff70e8fd32101fc95a060&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/50848b17-5d32-42ad-9219-cce4e7fc31e7/license?refs=542cfff737bcd79bb89ff70e8fd32101fc95a060&source=prChecksComment)

adamraya committed:
test: lower bundle size limit to test CI failure

This is a minimal change to verify the new bundle-size workflow
correctly fails when bundle size exceeds limits. Will be reverted.

adamraya committed:
revert: restore original bundle size limit to 62 kB

Reverts the test change that lowered the bundle size limit from 62 kB to 50 kB.
The CI workflow decoupling test was successful.

adamraya committed:
test: add unused variable to trigger lint error

Added an unused variable to test that the lint workflow fails correctly
while the test workflow continues to run independently.

adamraya committed:
fix: remove test lint error

Removed the unused variable that was added to test lint workflow failure.
The lint workflow decoupling test was successful.

alexvuong commented on .github/workflows/bundle-size.yml:
```diff
@@ -0,0 +1,38 @@
+# you must check ${{ env.DEVELOP == 'true' }}. There's probably a better way to DRY,
+# but this is what we have for now.
+
+name: SalesforceCommerceCloud/pwa-kit/bundle-size
```
I checked the CI flow but could not find this flow. Do I miss something?

alexvuong replied:
oh sorry, I got it now. nvm. my bad

alexvuong commented:
changes look good. One question, since now we are breaking bundle size and lint flows out, meaning we need to generate projects to run these flow. How much time do we add up to current CI run?

adamraya committed:
test: add undefined variable to trigger actual lint error

Added undefined variable reference that causes ESLint 'no-undef' error,
not just a warning. This should cause the lint workflow to fail.

adamraya committed:
revert: restore main.jsx to original clean state

Removed both the unused variable and undefined variable that were
added for testing lint workflow failure. File is now back to original state.

adamraya commented:
> changes look good. One question, since now we are breaking bundle size and lint flows out, meaning we need to generate projects to run these flow. How much time do we add up to current CI run?

The total time to complete all the workflows is actually reduced. The main test workflow is now faster because we removed the lint and bundle size steps from it. The new separate workflows run at the same time (in parallel), so they don't add extra waiting time.

Before time to complete all workflows: ~25-35 minutes (one big test workflow doing everything step by step)
After time to complete all workflows: ~20-30 minutes (three workflows are running at the same time in parallel, we wait for the longest one to finish)

Test workflow: ~20-30 minutes (longest)
Lint workflow: ~15-20 minutes 
Bundle size workflow: ~5-8 minutes

adamraya committed:
clean: remove unused IS_NOT_FORK env variable from bundle-size workflow

The IS_NOT_FORK environment variable was defined but never used in any
conditional statements within the bundle-size workflow. Removing it
simplifies the workflow and allows bundle size checks to run on all
PRs including those from forks, which is beneficial for code quality.

adamraya committed:
clean: remove unused IS_NOT_FORK env variable from lint workflow

The IS_NOT_FORK environment variable was defined but never used in any
conditional statements within the lint workflow. Removing it simplifies
the workflow and allows linting checks to run on all PRs including
those from forks, which is beneficial for code quality.

alexvuong reviewed (approved):
LGTM
